silk/datadog_checks/silk/events.py

```diff
@@ -0,0 +1,28 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+ALERT_TYPES = {"INFO": "info", "ERROR": "error", "WARNING": "warning", "CRITICAL": "error"}
+
+
+class SilkEvent(object):
+    def __init__(self, raw_event, tags):
+        self.raw_event = raw_event
+
+        if self.raw_event:
+            self.payload = {
+                "timestamp": self.raw_event.get("timestamp"),
+                "event_type": self.raw_event.get("labels"),
+                "source_type_name": 'silk',
+                "alert_type": ALERT_TYPES[self.raw_event.get("level")],
+                "tags": tags[:],
+            }
+        else:
+            self.payload = None
```

Comment:
What is the need for the creation of `self.payload` here? Could we instead just do all of this transformation in `get_datadog_payload`?

Reply:
I originally modeled the events implementation after [VSphere's](https://github.com/DataDog/integrations-core/blob/cfd04ffcaed83b58f501d54e575371228ed727ed/vsphere/datadog_checks/vsphere/event.py#L42-L85), which split them up to do extra transformations. I wasn't sure if Silk would also have more transformations later down the line, so I kept the structure of VSphere. Should I just do the entire creation of the payload within `get_datadog_payload` in this case?

Reply:
I think it makes sense to create the payload in one place. Vsphere seems to be doing a lot of transformations while Silk doesn't have any now. I think if Silk were to need more transformations, it would be simple enough to add them if we create the whole payload at once, as well as more readable.